test/unit/bucket/test_bucket.py

```diff
@@ -2289,6 +2291,26 @@ def _check_local_file_contents(self, path, expected_contents):
             contents = f.read()
             self.assertEqual(contents, expected_contents)
 
+    @pytest.mark.apiver(from_ver=2)
+    def test_download_to_non_seekable_file(self):
+        file_version = self.bucket.upload_bytes(self.DATA.encode(), 'file1')
+
+        non_seekable_strategies = [
+            strat for strat in self.bucket.api.services.download_manager.strategies
+            if not isinstance(strat, ParallelDownloader)
+        ]
+        context = contextlib.nullcontext() if non_seekable_strategies else pytest.raises(
+            ValueError,
+            match='no strategy suitable for download was found!',
+        )
```

Comment:
I think there should always be at least one strategy to non-seekable files (otherwise the whole download to stdout won't work, right?), so I would write it like this:
```python
assert non_seekable_strategies, \
    'At least one seekable strategy should be available'
self.download_file_by_id(...)
assert OUTPUT == DATA
```
Or, maybe, leave current test as-is but make the assertion above a separate test? Wdyt?

Reply:
so what you see here is one of many tests cases which are tested against multiple scenarios using different strategy sets, and some of them explicitly focus on parallel downloader-only, in which we expect the failure (thus pytest.raises

testing, what is the original scenario assumptions doesn't make much sense IMO